pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentSubscription.java

```diff
@@ -1201,6 +1203,21 @@ public long estimateBacklogSize() {
     }
 
     public SubscriptionStatsImpl getStats(GetStatsOptions getStatsOptions) {
+        // So far, there is no case hits this check.
+        if (getStatsOptions.isGetEarliestTimeInBacklog()) {
+            throw new IllegalArgumentException("Calling the sync method subscription.getStats with"
```

Comment:
This check will break the pulsar running. 

Do you consider `@Deprecated`?

Reply:
See https://github.com/apache/pulsar/blob/9da7cbda676a65a608c3eb1ca001743a1675355f/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java#L1409

Reply:
> This check will break the pulsar running.

No case will encounter this check, so it is fine

> Do you consider @Deprecated?

Added

Reply:
I suggest you remove this check, and it is unecessary.

Reply:
> I suggest you remove this check, and it is unnecessary.

No, it can prevent in-correct usage in the future

Reply:
`@Deprecated` has been added, why add this check, we will clean up this method in the future.

Reply:
> @Deprecated has been added, why add this check, we will clean up this method in the future.
> IDE can help us to check the @Depercated.

It prevents PR containing incorrect usage from being merged without adequate review, so it is needed.